compiler_opt/tools/extract_ir.py

```diff
@@ -333,13 +336,23 @@ def main(argv):
   with multiprocessing.Pool(FLAGS.num_workers) as pool:
     relative_output_paths = pool.map(extract_artifacts, objs)
 
-    # Write all Non-None relative paths to FLAGS.output_dir/module_paths.
-    with open(
-        os.path.join(FLAGS.output_dir, 'module_paths'), 'w',
-        encoding='utf-8') as f:
-      for path in relative_output_paths:
-        if path is not None:
-          f.write(path + '\n')
+  # This comes first rather than later so global_command_override is at the top
+  # of the .json after being written
+  if FLAGS.thinlto_build == 'local':
+    corpus_description = {'global_command_override': UNSPECIFIED_OVERRIDE}
```

Comment:
when this will be set the true override commandline? am I missing anything or this is going to be implemented in the next patch?

Reply:
the user needs to set it in the .json file, I could add an input prompt right here though (which if left blank will default to unspecified)

Reply:
do you mean that the user need to manually set it in the json file after running the script?

Reply:
yes

Reply:
where does this command line come from? why we cannot have it in the script?

Reply:
the script is run on a backend-invoked-thinlto corpus (so ThinLTO is invoked by lld, rather than clang). We have a command to re-run lld, but it only works correctly if we want to train on the entire corpus on every iteration. (which takes ~3 minutes, so it's not practical)

There's currently no way to run training (module-by-module) via lld (looked into it, there's potentially one way, but it's broken), so the command to re-run lld is effectively useless. We need a command to run clang, to mimic how lld sets up the optimization pipeline, and that's left to the user to decide, since there's too many options to parse in the lld command.  Maybe in the future we can figure out what a "good approximation" is based solely off the lld command, but even this is subject to change at any time (when someone adds new features for example)